digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala

```diff
@@ -140,7 +140,9 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
     if (linkIds.nonEmpty) {
       if (frozenTimeVVHAPIServiceEnabled) {
         vvhClient.frozenTimeRoadLinkData.fetchByLinkIds(linkIds)
-      } else vvhClient.roadLinkData.fetchByLinkIds(linkIds)
+      } else {
+        vvhClient.roadLinkData.fetchByLinkIds(linkIds) ++ vvhClient.complementaryData.fetchByLinkIds(linkIds)

```

Comment:
You can do both with async calls using `fetchByLinkIdsF` and then wait. We also have this method `fetchVVHRoadLinksAndComplementaryFromVVH` improved to be async and use it here.